viewing_party/party.py

```diff
@@ -0,0 +1,199 @@
+            for friend_movie in friend_movies_list:
+                if friend_movie["title"] == title:
+                    unique = False
+                    break
+            if not unique:
+                break
+        if unique:
+            unique_movies.append(movie)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
+            if unique and friend_movie not in unique_movies:
+                unique_movies.append(friend_movie)
+
+    return unique_movies
+
+def get_available_recs(user_data):
+    """user_data
+    {    subscriptions["name, name , name"]}
+    """
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
+            if unique and friend_movie not in unique_movies and friend_movie["host"] in user_data["subscriptions"]:
+                unique_movies.append(friend_movie)
+
+    return unique_movies
+
+
+def get_new_rec_by_genre(user_data):
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+    genre = get_most_watched_genre(user_data)
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
+            if unique and friend_movie not in unique_movies and friend_movie["genre"] == genre:
+                unique_movies.append(friend_movie)
+
+    return unique_movies
+
+def get_rec_from_favorites(user_data):
+    unique_movies = []
+
+    movie_list = user_data["favorites"]
+    friends_list = user_data["friends"]
+
+    for movie in movie_list:
+        title = movie["title"]
+        unique = True
+        for friend in friends_list:
+            friend_movies_list = friend["watched"]
+            for friend_movie in friend_movies_list:
+                if friend_movie["title"] == title:
+                    unique = False
+                    break
+            if not unique:
+                break
```

Comment:
Notice how similar this code is to getting the unique movies not watched by the user's friends. Could we use that function to simplify this code?